src/iqm/qiskit_iqm/iqm_provider.py

```diff
@@ -240,49 +241,103 @@ def serialize_circuit(self, circuit: QuantumCircuit) -> Circuit:
+                native_inst = Instruction(name='prx', qubits=qubit_names, args={'angle_t': 0.5, 'phase_t': 0.25})
             elif instruction.name == 'ry':
                 angle_t = float(instruction.params[0] / (2 * np.pi))
-                instructions.append(
-                    Instruction(name='prx', qubits=qubit_names, args={'angle_t': angle_t, 'phase_t': 0.25})
-                )
+                native_inst = Instruction(name='prx', qubits=qubit_names, args={'angle_t': angle_t, 'phase_t': 0.25})
             elif instruction.name == 'cz':
-                instructions.append(Instruction(name='cz', qubits=qubit_names, args={}))
+                native_inst = Instruction(name='cz', qubits=qubit_names, args={})
             elif instruction.name == 'move':
-                instructions.append(Instruction(name='move', qubits=qubit_names, args={}))
+                native_inst = Instruction(name='move', qubits=qubit_names, args={})
             elif instruction.name == 'barrier':
-                instructions.append(Instruction(name='barrier', qubits=qubit_names, args={}))
+                native_inst = Instruction(name='barrier', qubits=qubit_names, args={})
             elif instruction.name == 'measure':
-                mk = MeasurementKey.from_clbit(clbits[0], circuit)
-                instructions.append(Instruction(name='measure', qubits=qubit_names, args={'key': str(mk)}))
+                clbit = circuit_instruction.clbits[0]  # always a single-qubit measurement
+                mk = str(MeasurementKey.from_clbit(clbit, circuit))
+                native_inst = Instruction(name='measure', qubits=qubit_names, args={'key': mk})
+                clbit_to_measure[clbit] = native_inst
+            elif instruction.name == 'reset':
+                # implemented using a measure instruction to measure the qubits, and
+                # one cc_prx per qubit to conditionally flip it to |0>
+                feedback_key = '_reset'
+                instructions.append(
+                    Instruction(
+                        name='measure',
+                        qubits=qubit_names,
+                        args={
+                            # HACK to get something unique, remove when key can be omitted
+                            'key': f'_reset_{len(instructions)}',
+                            'feedback_key': feedback_key,
+                        },
+                    )
+                )
+                for q in qubit_names:
+                    physical_qubit_name = self._idx_to_qb[int(q)]
+                    instructions.append(
+                        Instruction(
+                            name='cc_prx',
+                            qubits=[q],
+                            args={
+                                'angle_t': 0.5,
+                                'phase_t': 0.0,
+                                'feedback_key': feedback_key,
```

Comment:
How does the QPU know which bit in the feedback_key measurement belongs to which qubit? We measure n qubits and cc_prx on 1 qubit n times. 
If you are not allowed to control from a measurement on another qubit, the documentation example would not work.

Reply:
`cc_prx` takes both a `feedback_key` and a `feedback_qubit`. The key indentifies the measurement op (there may be multiple), and the qubit identifies the qubit.

Reply:
Ah I see. So if I wanted to mix up the reset measurements and their X gates, I would shuffle the `feedback_qubit`s not the `feedback_key`. That makes sense.